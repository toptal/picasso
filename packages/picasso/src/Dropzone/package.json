{
  "name": "@toptal/picasso-dropzone",
  "version": "1.0.0",
  "main": "index.js",
  "dependencies": {
    "@toptal/picasso-icon": "^1.0.0",
    "@toptal/picasso-form-hint": "^1.0.0",
    "@toptal/picasso-form-error": "^1.0.0",
    "@toptal/picasso-container": "^1.0.0",
    "@toptal/picasso-file-list": "^1.0.0",
    "@toptal/picasso-typography": "^1.0.0",
    "@toptal/picasso-utils/use-deprecation-warnings": "^1.0.0"
  }
}