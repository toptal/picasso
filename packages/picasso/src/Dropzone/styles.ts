import type { Theme } from '@material-ui/core/styles'
import { createStyles } from '@material-ui/core/styles'

export default ({ palette, sizes, transitions }: Theme) =>
  createStyles({
    root: {
      padding: '20px',
      borderWidth: sizes.borderWidth,
      borderRadius: sizes.borderRadius.medium,
      borderColor: palette.grey.light2,
      borderStyle: 'dashed',
      backgroundColor: palette.common.white,
      color: palette.grey.dark,
      outline: 'none',
      transition: `all ${transitions.duration.short}ms ${transitions.easing.easeOut}`,
      transitionProperty: 'border, background-color',
      gap: '0.5rem',
      '&:hover, &$hovered, &:focus, &$focused, &$dragActive': {
        borderColor: palette.blue.main,
        cursor: 'pointer',
      },
      '&$disabled': {
        backgroundColor: palette.grey.lighter,
        '&:hover': {
          cursor: 'no-drop',
          borderColor: palette.grey.light2,
        },
      },
    },
    hint: {
      margin: 0,
      '& > *': {
        lineHeight: '1rem',
      },
    },
    completed: {},
    error: {
      margin: 0,
    },
    dragActive: {},
    hovered: {},
    disabled: {},
    focused: {},
  })
