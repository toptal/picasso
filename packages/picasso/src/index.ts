export { BaseProps } from '@toptal/picasso-shared'
export type { ColorType, SpacingType, SizeType } from '@toptal/picasso-shared'
export { AccordionCompound as Accordion } from './AccordionCompound'
export type { AccordionProps } from './Accordion'
export { default as AccountSelect } from './AccountSelect'
export type { AccountSelectProps } from './AccountSelect'
export { default as Amount } from './Amount'
export type { AmountProps } from './Amount'
export { default as Autocomplete } from './Autocomplete'
export type {
  AutocompleteProps,
  Item as AutocompleteItem,
} from './Autocomplete'
export { AvatarCompound as Avatar } from './AvatarCompound'
export type { AvatarProps } from './Avatar'
export { ButtonCompound as Button } from './ButtonCompound'
export type { ButtonProps, VariantType as ButtonVariantType } from './Button'
export type { ButtonCheckboxProps } from './ButtonCheckbox'
export type { ButtonRadioProps } from './ButtonRadio'
export { CheckboxCompound as Checkbox } from './CheckboxCompound'
export type { CheckboxProps } from './Checkbox'
export type { CheckboxGroupProps } from './CheckboxGroup'
export { default as Container } from './Container'
export type { ContainerProps } from './Container'
export { DropdownCompound as Dropdown } from './DropdownCompound'
export type { DropdownProps } from './Dropdown'
export { default as DropdownArrow } from './DropdownArrow'
export type { DropdownArrowProps } from './DropdownArrow'
export { default as EnvironmentBanner } from './EnvironmentBanner'
export type { EnvironmentBannerProps } from './EnvironmentBanner'
export { default as FileInput } from './FileInput'
export type { FileInputProps } from './FileInput'
export { default as FileList } from './FileList'
export type { FileListProps } from './FileList'
export { FormCompound as Form } from './FormCompound'
export type { FormProps } from './Form'
export { default as FormHint } from './FormHint'
export type { FormHintProps } from './FormHint'
export { default as FormError } from './FormError'
export type { FormErrorProps } from './FormError'
export { default as FormLabel } from './FormLabel'
export type { FormLabelProps, RequiredDecoration } from './FormLabel'
export { GridCompound as Grid } from './GridCompound'
export type { GridProps, GridSize } from './Grid'
export { HelpboxCompound as Helpbox } from './HelpboxCompound'
export type { HelpboxProps } from './Helpbox'
export { default as Image } from './Image'
export type { ImageProps } from './Image'
export { default as Indicator } from './Indicator'
export type { IndicatorProps } from './Indicator'
export { default as Input } from './Input'
export type { InputProps } from './Input'
export { default as InputAdornment } from './InputAdornment'
export type { InputAdornmentProps } from './InputAdornment'
export { TagCompound as Tag } from './TagCompound'
export type { TagProps } from './Tag'
export type { TagRectangularProps } from './TagRectangular'
export { default as Link } from './Link'
export type { LinkProps } from './Link'
export { default as Loader } from './Loader'
export type { LoaderProps } from './Loader'
export { default as Logo } from './Logo'
export type { LogoProps } from './Logo'
export { MenuCompound as Menu } from './MenuCompound'
export type { MenuProps } from './Menu'
export { default as MenuItem } from './MenuItem'
export type { MenuItemProps } from './MenuItem'
export { ModalCompound as Modal } from './ModalCompound'
export type { ModalProps } from './Modal'
export { default as MonthSelect } from './MonthSelect'
export type { MonthSelectProps } from './MonthSelect'
export { NotificationCompound as Notification } from './NotificationCompound'
export type { NotificationProps } from './Notification'
export { default as NotificationActions } from './NotificationActions'
export type { NotificationActionsProps } from './NotificationActions'
export { default as NumberInput } from './NumberInput'
export type { NumberInputProps } from './NumberInput'
export { PageCompound as Page } from './PageCompound'
export type { PageProps } from './Page'
export { default as Pagination } from './Pagination'
export type { PaginationProps } from './Pagination'
export { default as Paper } from './Paper'
export type { PaperProps } from './Paper'
export { default as PromptModal } from './PromptModal'
export type { PromptModalProps } from './PromptModal'
export { RadioCompound as Radio } from './RadioCompound'
export type { RadioProps } from './Radio'
export type { RadioGroupProps } from './RadioGroup'
export { default as RichTextEditor } from './RichTextEditor'
export type {
  RichTextEditorProps,
  RichTextEditorChangeHandler,
} from './RichTextEditor'
export { default as Select } from './Select'
export type { SelectProps } from './Select'
export type {
  Option as SelectOption,
  ValueType as SelectValueType,
} from './SelectBase'
export type { PageSidebarProps } from './PageSidebar'
export { default as Slider } from './Slider'
export type { SliderProps } from './Slider'
export { default as ShowMore } from './ShowMore'
export type { ShowMoreProps } from './ShowMore'
export { StepperCompound as Stepper } from './StepperCompound'
export type { StepperProps } from './Stepper'
export type { StepperVerticalProps } from './StepperVertical'
export { TableCompound as Table } from './TableCompound'
export type { TableProps } from './Table'
export { default as TableFooter } from './TableFooter'
export type { TableFooterProps } from './TableFooter'
export { TabsCompound as Tabs } from './TabsCompound'
export type { TabsProps } from './Tabs'
export { TagSelectorCompound as TagSelector } from './TagSelectorCompound'
export type { TagSelectorProps } from './TagSelector'
export { default as Tooltip } from './Tooltip'
export type { TooltipProps } from './Tooltip'
export { default as Typography } from './Typography'
export type { TypographyProps } from './Typography'
export { default as UserBadge } from './UserBadge'
export type { UserBadgeProps } from './UserBadge'
export { default as YearSelect } from './YearSelect'
export type { YearSelectProps } from './YearSelect'
export { default as ApplicationUpdateNotification } from './ApplicationUpdateNotification'
export type { ApplicationUpdateNotificationProps } from './ApplicationUpdateNotification'
export { AlertCompound as Alert } from './AlertCompound'
export type { AlertProps } from './Alert'
export type { AlertInlineProps } from './AlertInline'
export { default as Switch } from './Switch'
export type { SwitchProps } from './Switch'
export { default as Quote } from './Quote'
export type { QuoteProps } from './Quote'
export { default as Rating } from './Rating'
export type { RatingStarsProps, RatingThumbsProps } from './Rating'
export {
  default as TreeView,
  TreeViewContainer,
  useTreeView,
  TreeNodeAvatar,
} from './TreeView'
export type { TreeNodeInterface } from './TreeView'
export { default as StaticTreeView } from './StaticTreeView'
export { default as TimePicker } from './TimePicker'
export type { TimePickerProps } from './TimePicker'
export { default as SkeletonLoader } from './SkeletonLoader'
export { default as PageHead } from './PageHead'
export type { PageHeadProps } from './PageHead'
export { default as ProgressBar } from './ProgressBar'
export type { ProgressBarProps } from './ProgressBar'
export { default as List } from './List'
export type { ListProps } from './List'
export { default as Badge } from './Badge'
export type { BadgeProps } from './Badge'
export { BreadcrumbsCompound as Breadcrumbs } from './BreadcrumbsCompound'
export type { BreadcrumbsProps } from './Breadcrumbs'
export { default as Calendar } from './Calendar'
export type { DateOrDateRangeType } from './Calendar'
export {
  default as DatePicker,
  datePickerParseDateString,
  DEFAULT_DATE_PICKER_DISPLAY_DATE_FORMAT,
  DEFAULT_DATE_PICKER_EDIT_DATE_FORMAT,
} from './DatePicker'
export type {
  DatePickerValue,
  DatePickerInputCustomValueParser,
  DatePickerInputCustomValueParserParameters,
  DatePickerProps,
} from './DatePicker'
export { default as Drawer } from './Drawer'
export type { DrawerProps } from './Drawer'
export { default as Dropzone, ErrorCode as DropzoneErrorCode } from './Dropzone'
export type {
  DropzoneProps,
  FileUpload as DropzoneFileUpload,
  FileError as DropzoneFileError,
  FileRejection as DropzoneFileRejection,
  DropEvent as DropzoneDropEvent,
} from './Dropzone'
export { OverviewBlockCompound as OverviewBlock } from './OverviewBlockCompound'
export type { OverviewBlockProps } from './OverviewBlock'
export { default as TypographyOverflow } from './TypographyOverflow'
export type { TypographyOverflowProps } from './TypographyOverflow'
export { default as EmptyState } from './EmptyState'
export { default as Section } from './Section'
export type { SectionProps } from './Section'
export { NoteCompound as Note } from './NoteCompound'
export type { NoteProps } from './Note'
export { default as Timeline } from './Timeline'
export type { TimelineProps } from './Timeline'
export { default as RichText } from './RichText'
export type { RichTextProps, ASTType } from './RichText'
export { default as PasswordInput } from './PasswordInput'
export type { PasswordInputProps } from './PasswordInput'
export { default as FieldRequirements } from './FieldRequirements'
export type {
  FieldRequirementsProps,
  FieldRequirement,
} from './FieldRequirements'
export type { Status as OutlinedInputStatus } from './OutlinedInput'
export { default as AvatarUpload } from './AvatarUpload'
export type {
  AvatarUploadProps,
  FileUpload as AvatarUploadFileUpload,
  FileError as AvatarUploadFileError,
  FileRejection as AvatarUploadFileRejection,
  DropEvent as AvatarUploadDropEvent,
} from './AvatarUpload'

// hygen code generator inserts export statements above this comment.
export * from './Icon'
