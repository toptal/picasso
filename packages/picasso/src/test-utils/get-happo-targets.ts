type Result = Record<string, any>

/**
 * Generates Happo targets for responsive visual tests
 *
 * @param screenWidths Screen widths to generate targets for
 * @returns {Result} Object with Happo targets
 */
export const getHappoTargets = (screenWidths: number[]) =>
  screenWidths.reduce<Result>((acc: Result, width: number) => {
    const name = `chrome-desktop-width-${width}`

    acc[name] = {
      name,
      browser: 'chrome',
      viewport: `${width}x1024`,
    }

    return acc
  }, {})
