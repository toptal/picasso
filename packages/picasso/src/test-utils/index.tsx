import type { ReactElement } from 'react'
import React from 'react'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'

import TestingPicasso from './TestingPicasso'

export type PicassoConfig = {
  titleCase: boolean
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  picassoConfig?: PicassoConfig
): RenderResult => {
  const { rerender, ...rest } = render(
    <TestingPicasso titleCase={picassoConfig?.titleCase}>{ui}</TestingPicasso>,
    options
  )

  const customRerender: typeof rerender = newUi =>
    rerender(
      <TestingPicasso titleCase={picassoConfig?.titleCase}>
        {newUi}
      </TestingPicasso>
    )

  return { rerender: customRerender, ...rest }
}

export * from '@testing-library/react'
export { customRender as render, TestingPicasso }
export { getCheckpoints } from './get-checkpoints'
