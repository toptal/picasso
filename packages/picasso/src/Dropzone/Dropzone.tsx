import React, { forwardRef } from 'react'
import { useDropzone } from 'react-dropzone'
import cx from 'classnames'
import { makeStyles, Theme } from '@material-ui/core/styles'
import { BaseProps } from '@toptal/picasso-shared'

import { Upload24 } from '../Icon'
import FormHint from '../FormHint'
import FormError from '../FormError'
import Container from '../Container'
import FileList from '../FileList'
import Typography from '../Typography'
import { FileUpload, DropzoneOptions } from './types'
import { usePropDeprecationWarning } from '../utils/use-deprecation-warnings'
import styles from './styles'

export interface Props extends BaseProps {
  /**
   * Set accepted file types. See https://github.com/okonet/attr-accept for more information.
   */
  accept?: DropzoneOptions['accept']
  /** Enable/disable the dropzone */
  disabled?: boolean
  /** Maximum file size (in bytes) */
  maxSize?: number
  /** Minimum file size (in bytes) */
  minSize?: number
  /** Allow drag 'n' drop (or selection from the file dialog) of multiple files */
  multiple?: boolean
  /** The text of the hint */
  hint?: string
  /** Hide/Show the content text */
  hideContentText?: boolean
  /** Callback invoked when a file item is removed */
  onRemove?: (fileName: string, index: number) => void
  /** Callback for when the drop event occurs */
  onDropAccepted?: DropzoneOptions['onDropAccepted']
  /** callback for when the drop event occurs */
  onDropRejected?: DropzoneOptions['onDropRejected']
  /** callback for when the drop event occurs */
  onDrop?: DropzoneOptions['onDrop']
  /** Custom validation function */
  validator?: DropzoneOptions['validator']
  /** Value uses the File interface. */
  value?: FileUpload[]
  /**
   * @deprecated **Use value[n].error instead.**
   * Provide reasons why files couldn't be dropped into dropzone
   */
  errorMessages?: string[]
  focused?: boolean
  hovered?: boolean
}

const useStyles = makeStyles<Theme>(styles, { name: 'Dropzone' })

export const Dropzone = forwardRef<HTMLInputElement, Props>(function Dropzone(
  props,
  ref
) {
  const {
    hint,
    hideContentText,
    onRemove,
    value,
    className,
    errorMessages = [],
    style,
    'data-testid': dataTestId,
    focused,
    hovered,

    // dropzoneOptions
    accept,
    minSize,
    maxSize,
    multiple,
    disabled,
    onDrop,
    onDropAccepted,
    onDropRejected,
    validator,
  } = props

  const isDisabled = Boolean(
    disabled || (!multiple && value && value.length > 0)
  )

  const { getRootProps, isDragActive, getInputProps } = useDropzone({
    accept,
    minSize,
    maxSize,
    multiple,
    disabled: isDisabled,
    onDrop,
    onDropAccepted,
    onDropRejected,
    validator,
  })

  const classes = useStyles()

  usePropDeprecationWarning({
    props,
    name: 'errorMessages',
    componentName: 'Dropzone',
    description:
      'Use the `value[n].error` prop instead. `errorMessages` is deprecated and will be removed in the next major release.',
  })

  return (
    <Container style={style} ref={ref} className={className}>
      <Container
        flex
        direction='column'
        alignItems='center'
        data-testid={dataTestId}
        {...getRootProps({
          className: cx(classes.root, {
            [classes.dragActive]: isDragActive,
            [classes.hovered]: hovered,
            [classes.disabled]: isDisabled,
            [classes.focused]: focused,
          }),
        })}
      >
        <input {...getInputProps({ className: classes.nativeInput })} />
        <Upload24 color='darkGrey' />
        {!hideContentText && (
          <Typography size='medium' color='black' weight='semibold'>
            Click or drag to upload
          </Typography>
        )}
        {hint && <FormHint className={cx(classes.hint)}>{hint}</FormHint>}
        {errorMessages.length > 0 &&
          errorMessages.map((error, index) => (
            <FormError
              className={classes.error}
              key={`${error}-${String(index)}`}
            >
              {error}
            </FormError>
          ))}
      </Container>
      {value && value.length > 0 && (
        <Container top='xsmall'>
          <FileList files={value} onItemRemove={onRemove} />
        </Container>
      )}
    </Container>
  )
})

Dropzone.displayName = 'Dropzone'

Dropzone.defaultProps = {
  disabled: false,
  maxSize: Infinity,
  minSize: 0,
  multiple: true,
}

export default Dropzone
