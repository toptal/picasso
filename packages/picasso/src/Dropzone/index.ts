import { OmitInternalProps } from '@toptal/picasso-shared'

import { Props } from './Dropzone'
export { default } from './Dropzone'
export * from './types'

export type DropzoneProps = OmitInternalProps<Props>
