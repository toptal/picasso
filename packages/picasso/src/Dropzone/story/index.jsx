import { Dropzone } from '../Dropzone'
import PicassoBook from '~/.storybook/components/PicassoBook'

const page = PicassoBook.section('Components').createPage(
  'Dropzone',
  `
  ${PicassoBook.createBaseDocsLink(
    'https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/58ee1df911f7d6ca2d9214171088a69bdff23401/files/13531207-e094-44ec-ae1f-f27628c1aea5/layers/27C9F1FA-E469-4AA4-936C-DD752F77ADE3?mode=build&selected=root-8BD3C500-A858-4CC4-8CE8-02E0453C0C4D&sha=58ee1df911f7d6ca2d9214171088a69bdff23401'
  )}
  `
)

page.createTabChapter('Props').addComponentDocs({
  component: Dropzone,
  name: 'Dropzone',
  additionalDocs: {
    accept: {
      name: 'accept',
      type: 'string | string[]',
      description:
        'Set accepted file types. See https://github.com/okonet/attr-accept for more information. Keep in mind that mime type determination is not reliable across platforms. CSV files, for example, are reported as text/plain under macOS but as application/vnd.ms-excel under Windows. In some cases there might not be a mime type set at all. See: https://github.com/react-dropzone/react-dropzone/issues/276'
    },
    onDropAccepted: {
      name: 'onDropAccepted',
      type: {
        name: 'function',
        description: '(files: File[], event) => void'
      },
      description:
        'Callback for when the drop event occurs. Note that if no files are accepted, this callback is not invoked.'
    },
    onDropRejected: {
      name: 'onDropRejected',
      type: {
        name: 'function',
        description: '(fileRejections: FileRejection[], event) => void'
      },
      description:
        'Callback for when the drop event occurs. Note that if no files are rejected, this callback is not invoked.'
    },
    validator: {
      name: 'validator',
      type: {
        name: 'function',
        description: '(file:File) => FileError | FileError[] | null'
      },
      description: 'Custom validation function',
      defaultvalue: 'null'
    },
    onDrop: {
      name: 'onDrop',
      type: {
        name: 'function',
        description:
          '(acceptedFiles: File[], fileRejections: FileRejection[], event:DragEvent | Event) => void'
      },
      description: `
Callback for when the drop event occurs. Note that this callback is invoked after the getFilesFromEvent callback is done.

Files are accepted or rejected based on the accept, multiple, minSize and maxSize props. accept must be a valid MIME type according to input element specification or a valid file extension. If multiple is set to false and additional files are dropped, all files besides the first will be rejected. Any file which does not have a size in the minSize, maxSize range, will be rejected as well.

Note that the onDrop callback will always be invoked regardless if the dropped files were accepted or rejected. If you'd like to react to a specific scenario, use the onDropAccepted/onDropRejected props.

onDrop will provide you with an array of File objects which you can then process and send to a server. For example, with SuperAgent as a http/ajax library:

      function onDrop(acceptedFiles) {
        const req = request.post('/upload')
        acceptedFiles.forEach(file => {
          req.attach(file.name, file)
        })
        req.end(callback)
      }
      `
    }
  }
})

page
  .createChapter()
  .addExample('Dropzone/story/Default.example.tsx', {
    title: 'Default',
    takeScreenshot: false
  })
  .addExample('Dropzone/story/Uploader.example.tsx', {
    title: 'With upload',
    takeScreenshot: false
  })
  .addExample('Dropzone/story/Uploading.example.tsx', {
    title: 'Uploading',
    takeScreenshot: false
  })
  .addExample('Dropzone/story/Disabled.example.tsx', {
    title: 'Completed',
    takeScreenshot: false
  })
  .addExample('Dropzone/story/Multiple.example.tsx', {
    title: 'Completed Multiple files',
    takeScreenshot: false
  })
  .addExample('Dropzone/story/Error.example.tsx', {
    title: 'Upload Failed',
    takeScreenshot: false
  })
  .addExample('Dropzone/story/SingleFile.example.tsx', {
    title: 'Single File',
    takeScreenshot: false
  })
