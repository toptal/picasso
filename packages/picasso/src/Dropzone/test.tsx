import React from 'react'
import { render } from '@toptal/picasso/test-utils'
import { OmitInternalProps } from '@toptal/picasso-shared'

import Dropzone, { Props } from './Dropzone'

const renderDropzone = (props: OmitInternalProps<Props>) =>
  render(<Dropzone {...props} />)

describe('Dropzone', () => {
  it('renders', () => {
    const { queryByText, container } = renderDropzone({})

    expect(queryByText('Click or drag file to upload')).toBeVisible()
    expect(container).toMatchSnapshot()
  })

  it('shows error', () => {
    const { queryByText } = renderDropzone({
      errorMessages: ['error example'],
      hint: 'hint example'
    })

    expect(queryByText('hint example')).not.toBeInTheDocument()
    expect(queryByText('error example')).toBeVisible()
  })

  it('renders hint', () => {
    const { queryByText } = renderDropzone({ hint: 'hint example' })

    expect(queryByText('hint example')).toBeVisible()
  })

  describe("when doesn't accept multiple files and there is already a file", () => {
    it('disables the dropzone', () => {
      const { getByTestId } = renderDropzone({
        'data-testid': 'dropzone',
        multiple: false,

        value: [
          {
            uploading: false,
            progress: 0,
            file: new File(['resume.pdf'], 'resume.pdf')
          }
        ]
      })

      expect(getByTestId('dropzone').className).toContain('disabled')
    })
  })
})
