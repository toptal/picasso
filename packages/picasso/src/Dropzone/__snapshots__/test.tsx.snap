// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropzone renders 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <div
      class=""
    >
      <div
        class="PicassoContainer-centerAlignItems PicassoContainer-flex PicassoContainer-column Dropzone-root"
        role="presentation"
        tabindex="0"
      >
        <input
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
        <svg
          class="PicassoSvgUpload24-root PicassoSvgUpload24-darkGrey"
          style="min-width: 24px; min-height: 24px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M2 18v3h19v-3h1v4H1v-4h1Zm9.5-16.207L16.207 6.5l-.707.707-3.501-3.501V19H11l-.001-15.294L7.5 7.207 6.793 6.5 11.5 1.793Z"
          />
        </svg>
        <p
          class="MuiTypography-root PicassoTypography-bodyMedium PicassoTypography-black PicassoTypography-semibold MuiTypography-body1"
        >
          Click or drag to upload
        </p>
      </div>
    </div>
  </div>
</div>
`;
